multicluster/controllers/multicluster/member/gateway_controller.go

```diff
@@ -77,7 +77,21 @@ func (r *GatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	var err error
 	var commonArea commonarea.RemoteCommonArea
 	commonArea, r.localClusterID, err = r.commonAreaGetter.GetRemoteCommonAreaAndLocalID()
+	gw := &mcv1alpha1.Gateway{}
+	var isNotFound bool
+	if err := r.Client.Get(ctx, req.NamespacedName, gw); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return ctrl.Result{}, err
+		}
+		isNotFound = true
+	}
 	if commonArea == nil {
```

Comment:
When `commonArea` is nil, should we always ignore any event? Do we resync after `commonArea` is initialized? Should we do that?

Do we have a way to start controllers after `commonArea` is initialized?

Reply:
Based on current implementation, we should retry to handle Gateway events to make sure the Gateway change is reflected to ClusterInfo of ResourceExport when the ClusterSet is created. A better way is to resync after commonArea is initialized. I will check how to refine this and may submit a separate PR.

I feel we should start controllers after commonArea is initialized, I need to dig more before confirming this is doable.

Reply:
Refined this part with error string check and more comments.

Reply:
Do you mean you would merge the current implementation first, and then do another PR to change to resync after commonArea initialization? Why that?

Reply:
I need to investigate how to do the resync, I was thinking it's easier to back-port if we have a separate PR. But maybe we can just keep the change in this release only.

Reply:
When will be the deadline for the fix to get into this release? I have troubles to understand the current implementation to review the PR.

Reply:
Our planed release date is Oct 27, so this PR should be in around Oct 25. I will check how to refine the member part, so the event can be handled only without retry when the ClusterSet(the common area) is ready.

For current implementation, all member controllers will start at once and reconcile resources events even there is no valid common area, this means the controller will re-queue and retry until the ClusterSet is created and a common area is ready.

Reply:
Refined member controllers to skip reconciling objects when there is no ClusterSet CR.